src/corehost/cli/fxr_resolver.cpp

```diff
@@ -54,80 +53,90 @@ namespace
     }
 }
 
-bool resolve_fxr_path(const pal::string_t& root_path, pal::string_t* out_dotnet_root, pal::string_t* out_fxr_path)
+bool fxr_resolver::try_get_path(const host_mode_t mode, const pal::string_t& root_path, pal::string_t* out_dotnet_root, pal::string_t* out_fxr_path)
 {
     pal::string_t fxr_dir;
-#if FEATURE_APPHOST || FEATURE_LIBHOST
```

Comment:
We go to great lengths to avoid breaking changes to the muxer.  I wonder if we should keep some #if here.  Something like:
```C++
#if !FEATURE_MUXER
```

Or maybe it would be better to leave the interface unchanged.  Keep most of the original code and interface. I see no significant functional change to this file, just refactoring for cleanup.
```C++
bool resolve_fxr_path(const pal::string_t& root_path, pal::string_t* out_dotnet_root, pal::string_t* out_fxr_path)
```

Reply:
I think I'm missing what the breaking change is here?

The functionality / logic should be the same. For nethost, we need to be able to resolve using either the muxer behaviour or the apphost/libhost behaviour, so I removed the preprocessor conditions here. But corehost.cpp is setting the mode based on the defines such that the behaviour should be the same (and com/ijwhost, which have FEATURE_LIBHOST, are always using host_mode_t::libhost)

Reply:
This isn't an exported function. There is no breaking change here.

Reply:
We spent months getting approval to make changes to the muxer.  We had to justify changes to 
* Add the manifest
* Add logging to file support
* Add support for SemVer 2.0.0

We could logically reason that none of these were breaking.

This PR is introducing a non-functional change to the muxer.  I agree it shouldn't be breaking, but it does change muxer code.  It is introducing a switch and dead code which was not compiled for the muxer before.

If there was a functional reason for the code change, it is OK to keep the switch. I would still think about restoring an `#if` to eliminate the dead code for the muxer. I'll let @vitek-karas decide if I am being overly cautious.

Reply:
>We spent months getting approval to make changes to the muxer.

Not for `comhost` which had much more impactful functional changes - the very notion of `libhost` for instance.

Reply:
Actually manifests have OS level requirements and impact, logging to file is an operational change since it has potential perf and file system impact, and versioning impacts literally everything. I really don't see how any of these code changes are even close to the same level as those.

Reply:
> Not for comhost which had much more impactful functional changes - the very notion of libhost for instance.

Only because we didn't mention they affected the muxer.  If there are changes to the muxer, we must review them carefully. @jeffschwMSFT wants us to make sure there is sufficient coverage to guarantee the muxer works properly for every shipping .NET Core version.

Reply:
>Only because we didn't mention they affected the muxer.

That just isn't true. The impact of `comhost` on the muxer was discussed extensively.

I agree that coverage is an important consideration. I am hoping the start of the native testing here will allow us to do just that. @vitek-karas also mentioned you had started work on some kind of native testing. Is your work applicable here?

Reply:
> That just isn't true

Offline

> Is your work applicable here?

The unit and E2E tests are checked in. In my opinion they were sufficient to test the SemVer 2.0.0 work.

Reply:
Since it turned out we didn't need the muxer behaviour in nethost after all, I went back to the preprocessor conditions.